freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserSyncService.java

```diff
@@ -283,6 +290,12 @@ private SyncStatusDetail internalSynchronizeStack(Stack stack, UmsUsersState ums
                 workloadCredentialService.setWorkloadCredentials(freeIpaClient, umsUsersState.getUsersWorkloadCredentialMap(), warnings::put);
             }
 
+            boolean hasCloudIdentityEntitlement = entitlementService.cloudIdentityMappingEnabled(INTERNAL_ACTOR_CRN, stack.getAccountId());
+            // TODO For now we only sync cloud ids during full sync. We should eventually allow more granular syncs (actor level and group level sync).
+            if (hasCloudIdentityEntitlement && fullSync) {
```

Comment:
nit: if you reverse the order of conditions and inline the entitlement check here, then we can skip the entitlement check if it's not a full sync.

Reply:
nice one